root/static/scripts/edit/URLCleanup.js

```diff
@@ -1448,19 +1448,24 @@ const CLEANUPS = {
   },
   'instagram': {
     match: [new RegExp('^(https?://)?([^/]+\\.)?instagram\\.com/', 'i')],
-    type: LINK_TYPES.socialnetwork,
+    type: _.defaults(
+      {},
+      LINK_TYPES.socialnetwork,
+      {recording: LINK_TYPES.streamingfree.recording},
+    ),
     clean: function (url) {
-      // Ignore explore/photo URLs since we'll block them anyway
+      url = url.replace(/^(?:https?:\/\/)?(?:[^\/]+\.)?instagram\.com\/(?:p|tv)\/([^\/?#]+).*$/, 'https://www.instagram.com/p/$1/');
+      // Ignore explore URLs since we'll block them anyway
       if (!(/^https:\/\/www\.instagram\.com\/(explore|p)\//.test(url))) {
         // Point /stories/ sections to the main user profile instead
         url = url.replace(/^(?:https?:\/\/)?(?:[^\/]+\.)?instagram\.com\/stories\/([^\/?#]+)\/?(?:[\/?#].*)?$/, 'https://www.instagram.com/$1/');
         url = url.replace(/^(?:https?:\/\/)?(?:[^\/]+\.)?instagram\.com\/([^\/?#]+)\/?(?:[\/?#].*)?$/, 'https://www.instagram.com/$1/');
       }
       return url;
     },
-    validate: function (url) {
+    validate: function (url, id) {
       // Block explore/photo URLs, which aren't really a social network link
-      if (/^https:\/\/www\.instagram\.com\/p\//.test(url)) {
+      if (/^https:\/\/www\.instagram\.com\/p\//.test(url) && id !== LINK_TYPES.streamingfree.recording) {
```

Comment:
Maybe also edit the error text to something like: 'Please do not link directly to images, link to the appropriate Instagram profile page instead. If you are trying to link to a video for a recording, use the "stream for free" relationship to do so.'?

Reply:
Good catch. If a /p/ URL is added to a recording, “stream for free” will be automatically selected, no error message will be shown. So this error message should be for other entity types. I suggest to create a standalone recording in commit https://github.com/metabrainz/musicbrainz-server/pull/1590/commits/435d70b91d726e263ccd591020d41fabe4a58b62.